src/test/tools/ddmap/ddmap.c

```diff
@@ -212,7 +250,39 @@ ddmap_write(const char *path, const char *str, off_t offset, size_t len)
 		if (ddmap_zero(path, (size_t)offset + length, len - length))
 			return -1;
 	}
+	return 0;
+}
+
+/*
+ * compute_checksum -- (internal) compute checksum of input file
+ */
+static int
+ddmap_checksum(const char *path, size_t len, off_t offset)
+{	
+	int sfd;
+	char *src;
+	uint64_t checksum = 0;
+	size_t filesize = (size_t)util_file_get_size(path);
+	
+	if (!util_file_is_device_dax(path)){
```

Comment:
Do we need to check this condition?
src = util_file_map_whole(path) should work.